### PR TITLE
Added ports to default configuration.

### LINKED ISSUE #6132: Add Redis to default list of Opaque ports
Redis is pretty common. We should add it to the list of default skip ports.

https://github.com/linkerd/linkerd2/blob/da588130544154fdb94afb64147fa745ce852fc4/charts/linkerd2/values.yaml#L110

### PR DESCRIPTION
**Pull request by JasonMorgan**
Default Linkerd skip and opaque port configuration

Missing default ports based on docs

Addressed: Add Redis to default list of Opaque ports #6132

Once merged, the default install values will match the recommendations in Linkerd's TCP ports guide.

Fixes #6132

### CODE REVIEW

**Comment by Pothulapati** [comment]
Hey 👋🏼 

I think you need to update the golden test files by running `go test -mod=readonly ./... --update` and also the helm docs by running `./bin/helm-docs` for the CI to go ✅

**Comment by JasonMorgan** [comment]
Roger that!

**Review by alpeb — COMMENTED** [review]
Unfortunately, there are a couple of manual updates required that are not cautght by the `-update` flag:
- `controller/proxy-injector/fake/data/pod.patch.json`
- The `expected` fixture in `pkg/charts/linkerd2/values_test.go`

**Comment by alpeb** [comment]
... and the `expectedInitContainer` fixture in `test/integration/inject/inject_test.go`

File: controller/proxy-injector/fake/data/pod.patch.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
         "--proxy-uid",
         "2102",
         "--inbound-ports-to-ignore",
-        "4190,4191"
+        "4190,4191,4567,4568"
```
Comment (alpeb) [inline controller/proxy-injector/fake/data/pod.patch.json:57]: You forgot to add `4567,4568`  to `--outbound-ports-to-ignore`:
```suggestion
        "--inbound-ports-to-ignore",
        "4190,4191,4567,4568",
        "--outbound-ports-to-ignore",
        "4567,4568"
```
I also realized `pod-with-debug.patch.json` and `pod-with-ns-annotations.patch.json` also require a similar update.

File: test/integration/inject/inject_test.go [hunk 2]
```diff
@@ -388,7 +388,7 @@ func TestInjectAutoPod(t *testing.T) {
 			"--outgoing-proxy-port", "4140",
 			"--proxy-uid", "2102",
 			// 1234,5678 were added at install time in `install_test.go`'s helmOverridesEdge()
-			"--inbound-ports-to-ignore", "4190,4191,1234,5678",
+			"--inbound-ports-to-ignore", "4190,4191,1234,5678,4567,4568",
```
Comment (alpeb) [inline test/integration/inject/inject_test.go:391]: and here:
```suggestion
			"--inbound-ports-to-ignore", "4190,4191,1234,5678,4567,4568",
         		"--outbound-ports-to-ignore", "4567,4568",
```

**Comment by JasonMorgan** [comment]
Ok, should have all those changes in. Sorry to be such a pain here!

**Comment by alpeb** [comment]
No problemo :+1: Although some of these new failures are very weird... investigating :mag:

**Comment by alpeb** [comment]
I might have provided you with bad advice above. Anyways, this patch should take care of all the failures (also please remember to take care of the DCO):
```diff
diff --git a/controller/proxy-injector/fake/data/pod-with-debug.patch.json b/controller/proxy-injector/fake/data/pod-with-debug.patch.json
index c1fa6105..08869b9a 100644
--- a/controller/proxy-injector/fake/data/pod-with-debug.patch.json
+++ b/controller/proxy-injector/fake/data/pod-with-debug.patch.json
@@ -170,7 +170,7 @@
         },
         {
           "name": "LINKERD2_PROXY_INBOUND_PORTS_DISABLE_PROTOCOL_DETECTION",
-          "value": "25,443,587,3306,5432,11211"
+          "value": "25,443,587,3306,4444,5432,6379,9300,11211"
         },
         {
           "name": "_pod_ns",
diff --git a/controller/proxy-injector/fake/data/pod-with-ns-annotations.patch.json b/controller/proxy-injector/fake/data/pod-with-ns-annotations.patch.json
index 72d6c491..b09dd9a8 100644
--- a/controller/proxy-injector/fake/data/pod-with-ns-annotations.patch.json
+++ b/controller/proxy-injector/fake/data/pod-with-ns-annotations.patch.json
@@ -66,7 +66,7 @@
         "--inbound-ports-to-ignore",
         "4190,4191,4567,4568",
         "--outbound-ports-to-ignore",
-        "4567,4568,34567"
+        "34567"
       ],
       "image": "cr.l5d.io/linkerd/proxy-init:v1.3.13",
       "imagePullPolicy": "IfNotPresent",
@@ -170,7 +170,7 @@
         },
         {
           "name": "LINKERD2_PROXY_INBOUND_PORTS_DISABLE_PROTOCOL_DETECTION",
-          "value": "25,443,587,3306,5432,11211"
+          "value": "25,443,587,3306,4444,5432,6379,9300,11211"
         },
         {
           "name": "_pod_ns",
diff --git a/pkg/charts/linkerd2/values_test.go b/pkg/charts/linkerd2/values_test.go
index 59073f57..0dfc7e3b 100644
--- a/pkg/charts/linkerd2/values_test.go
+++ b/pkg/charts/linkerd2/values_test.go
@@ -86,6 +86,8 @@ func TestNewValues(t *testing.T) {
                        Await:                  true,
                },
                ProxyInit: &ProxyInit{
+                       IgnoreInboundPorts:  "4567,4568",
+                       IgnoreOutboundPorts: "4567,4568",
                        Image: &Image{
                                Name:    "cr.l5d.io/linkerd/proxy-init",
                                Version: testVersion,
diff --git a/test/integration/inject/inject_test.go b/test/integration/inject/inject_test.go
index 5ca32b6e..de2fb644 100644
--- a/test/integration/inject/inject_test.go
+++ b/test/integration/inject/inject_test.go
@@ -388,8 +388,8 @@ func TestInjectAutoPod(t *testing.T) {
                        "--outgoing-proxy-port", "4140",
                        "--proxy-uid", "2102",
                        // 1234,5678 were added at install time in `install_test.go`'s helmOverridesEdge()
-                       "--inbound-ports-to-ignore", "4190,4191,1234,5678,4567,4568",
-                       "--outbound-ports-to-ignore", "4567,4568",
+                       "--inbound-ports-to-ignore", "4190,4191,1234,5678",
+                       "--outbound-ports-to-ignore", "4567,4568",
                },
                Resources: v1.ResourceRequirements{
                        Limits: v1.ResourceList{
```

**Comment by JasonMorgan** [comment]
Forgot DCO on that last one. for the record: I agree to the DCO for all the commits in this PR.

**Review by adleong — APPROVED** [review]

**Review by kleimkuhler — APPROVED** [review]
Thanks @JasonMorgan!

**Review by alpeb — APPROVED** [review]

**Pull request merged**